staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -697,6 +705,60 @@ func (s *Scheme) addObservedVersion(version schema.GroupVersion) {
 	s.observedVersions = append(s.observedVersions, version)
 }
 
+func objectAPILifecycle(obj Object) (schema.APILifecycle, bool) {
+	hasLifecycle := false
+	ret := schema.APILifecycle{}
+	introduced, isIntroduced := obj.(apiLifecycleIntroduced)
+	if isIntroduced {
+		major, minor := introduced.APILifecycleIntroduced()
+		ret.IntroducedVersion = version.MajorMinor(uint(major), uint(minor))
+		hasLifecycle = true
+	}
+	removed, isRemoved := obj.(apiLifecycleRemoved)
+	if isRemoved {
+		major, minor := removed.APILifecycleRemoved()
+		ret.RemovedVersion = version.MajorMinor(uint(major), uint(minor))
+		hasLifecycle = true
+	}
+
+	return ret, hasLifecycle
+}
+
+type apiLifecycleIntroduced interface {
+	APILifecycleIntroduced() (major, minor int)
+}
+
+type apiLifecycleRemoved interface {
+	APILifecycleRemoved() (major, minor int)
+}
```

Comment:
These signatures surprised me, why not return a version type and let caller extract the major or minor as they need it?   Returning the whole version can have advantages like making comparisons easier?

Reply:
These interfaces already exist for lifecycle tags:
https://github.com/kubernetes/kubernetes/blob/445869a59bdbd1c587b72b52c5da94c1d1c316a1/staging/src/k8s.io/api/networking/v1alpha1/types.go#L26

Reply:
What I was wondering is why use a return type of `(major, minor int)` instead of returning a type the represents a version?

Reply:
That was added 4 years ago https://github.com/kubernetes/kubernetes/commit/4b0080d71f0b34a6793d9ea503b10cca01dc9fe6. 
We should probably change it to version, but I don't think we should change it in this PR.